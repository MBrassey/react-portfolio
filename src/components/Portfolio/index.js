import React, { useState } from "react";
import Project from "../Project";

function Portfolio() {
  const [projects] = useState([
    {
      name: "waviii.io",
      description:
        "Fully Decentralized ERC-20 Token, Wallet, Exchange & Price Chart - React, Web3js & RESTful API's ",
      link: "https://waviii.io",
      repo: "https://github.com/MBrassey/waviii.io",
    },
    {
      name: "BrewHound",
      description:
        "MERN stack Brewery Finder with GraphQL integration using multiple external API's",
      link: "https://brewhound-2.herokuapp.com",
      repo: "https://github.com/MBrassey/brewhound-2",
    },
    {
      name: "ALEXANDRIA",
      description:
        "MERN (MongoDB, Express.js, React, and Node.js) stack book search & save with GraphQL",
      link: "https://mbrassey-alexandria.herokuapp.com",
      repo: "https://github.com/MBrassey/alexandria",
    },
    {
      name: "TossUp",
      description:
        "Fully decentralized betting platform using Chainlink randomness, Ethereum Smartcontracts, IPFS & React frontend",
      link: "https://mbrassey-toss-up.on.fleek.co",
      repo: "https://github.com/MBrassey/toss-up",
    },
    {
      name: "Deep",
      description:
        "MERN (MongoDB, Express.js, React, and Node.js) stack social network ",
      link: "https://mbrassey-deep-thoughts.herokuapp.com",
      repo: "https://github.com/MBrassey/deep-thoughts",
    },
    {
      name: "waviii-token",
      description:
        "Ethereum Mainnet ERC-20 Token, total supply 1 million waviii. The token was built with Openzeppelin and deployed with Truffle.",
      link:
        "https://etherscan.io/token/0xBA00868912Af1a409F11E9c2B5d3a9376Cb3C2E2",
      repo: "https://github.com/MBrassey/waviii-token",
    },
    {
      name: "WeatherDash",
      description:
        "Weather dashboard concept using OpenWeather API, ipinfo.io API & jQuery/JS/HTML/CSS",
      link: "https://mbrassey.github.io/OpenWeather-Dashboard",
      repo: "https://github.com/MBrassey/OpenWeather-Dashboard",
    },
    {
      name: "TurqMelon",
      description:
        "Image ranking & tipping platform written in ReactJS, powered by Solidity Smart Contracts & IPFS",
      link: "https://github.com/MBrassey/TurquoiseMelon",
      repo: "https://github.com/MBrassey/TurquoiseMelon",
    },
    {
      name: "UpSpeak",
      description:
        "Ethereum Powered Decentralized Social Network & Free Speech Platform",
      link: "https://github.com/MBrassey/upSpeak",
      repo: "https://github.com/MBrassey/upSpeak",
    },
    {
      name: "TheDevBlog",
      description:
        "Full stack CMS type blog application utilizing ORM with Sequelize/MySQL and adhering to the Model-View-Controller paradigm",
      link: "https://thedevblog-mbrassey.herokuapp.com",
      repo: "https://github.com/MBrassey/TheDevBlog",
    },
    {
      name: "galler3",
      description:
        "Single-page photography portfolio application that uses React state, props, Hooks, and reusable UI components",
      link: "https://mbrassey.github.io/galler3",
      repo: "https://github.com/MBrassey/galler3",
    },
    {
      name: "ThoughtAPI",
      description:
        "API for a social network web application where users can share their thoughts, react to friendsâ€™ thoughts, and create a friend list",
      link: "https://thoughtstreamapi-mbrassey.herokuapp.com/api/users",
      repo: "https://github.com/MBrassey/ThoughtStreamAPI",
    },
    {
      name: "UltraQueue",
      description:
        "Personal kanban style queue that saves tasks to localstorage, packaged by @electron-forge for Windows, Mac & Linux",
      link: "https://mbrassey.github.io/UltraQueue",
      repo: "https://github.com/MBrassey/UltraQueue",
    },
    {
      name: "TechNewz",
      description:
        "Full stack Hacker News type blog application utilizing ORM with Sequelize and adhering to the Model-View-Controller paradigm",
      link: "https://technewz-mbrassey.herokuapp.com",
      repo: "https://github.com/MBrassey/TechNewz",
    },
    {
      name: "Ecommerce",
      description:
        "E-Commerce backend with Express.js and SQL database. Including robust REST API with multiple endpoints/routes",
      link: "https://ecommercesite-mbrassey.herokuapp.com/api/categories",
      repo: "https://github.com/MBrassey/EcommerceSite",
    },
    {
      name: "GitQuery",
      description:
        "Query the Github API for issues, usernames & languages with jQuery/JS/HTML/CSS",
      link: "https://mbrassey.github.io/GitQuery",
      repo: "https://github.com/MBrassey/GitQuery",
    },
    {
      name: "PassGen",
      description:
        "Configurable strong password generator written in pure JS/HTML/CSS",
      link: "https://mbrassey.github.io/PasswordGenerator",
      repo: "https://github.com/MBrassey/PasswordGenerator",
    },
    {
      name: "Scheduler",
      description:
        "Day scheduler concept using jQuery/JS/HTML/CSS, packaged by @electron-forge for Windows, Mac & Linux",
      link: "https://mbrassey.github.io/Scheduler",
      repo: "https://github.com/MBrassey/Scheduler",
    },
  ]);

  return (
    <div>
      <div className="flex-row">
        {projects.map((project, idx) => (
          <Project project={project} key={"project" + idx} />
        ))}
      </div>
    </div>
  );
}

export default Portfolio;
