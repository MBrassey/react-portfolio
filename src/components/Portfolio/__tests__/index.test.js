import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';

import Portfolio from '../';

afterEach(cleanup);

describe('Portfolio component', () => {
  it('renders', () => {
    render(<Portfolio />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Portfolio />);

    expect(asFragment()).toMatchSnapshot();
  });
})
