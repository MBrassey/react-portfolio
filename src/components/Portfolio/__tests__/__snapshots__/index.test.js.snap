// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex-row"
    >
      <div
        class="project"
      >
        <img
          alt="galler3"
          class="project-bg"
          src="galler3.png"
        />
        <div
          class="project-text"
        >
          <h3>
            <a
              href="https://mbrassey.github.io/galler3"
            >
              galler3
            </a>
              
            <a
              href="https://github.com/MBrassey/galler3"
            >
              <i
                class="fab fa-github"
              />
            </a>
          </h3>
          <p>
            Single-page photography portfolio application that uses React state, props, Hooks, and reusable UI components
          </p>
        </div>
      </div>
      <div
        class="project"
      >
        <img
          alt="waviii"
          class="project-bg"
          src="waviii.png"
        />
        <div
          class="project-text"
        >
          <h3>
            <a
              href="https://waviii.herokuapp.com/admin/Wallet"
            >
              waviii
            </a>
              
            <a
              href="https://github.com/MBrassey/waviii.io"
            >
              <i
                class="fab fa-github"
              />
            </a>
          </h3>
          <p>
            Web3 Ethereum Wallet & ERC-20 Token Swap React Components Inside Social Media Dapp Concept
          </p>
        </div>
      </div>
      <div
        class="project"
      >
        <img
          alt="TurqMelon"
          class="project-bg"
          src="TurqMelon.png"
        />
        <div
          class="project-text"
        >
          <h3>
            <a
              href="https://github.com/MBrassey/TurquoiseMelon"
            >
              TurqMelon
            </a>
              
            <a
              href="https://github.com/MBrassey/TurquoiseMelon"
            >
              <i
                class="fab fa-github"
              />
            </a>
          </h3>
          <p>
            Image ranking & tipping platform written in ReactJS, powered by Solidity Smart Contracts & IPFS
          </p>
        </div>
      </div>
      <div
        class="project"
      >
        <img
          alt="WeatherDash"
          class="project-bg"
          src="WeatherDash.png"
        />
        <div
          class="project-text"
        >
          <h3>
            <a
              href="https://mbrassey.github.io/OpenWeather-Dashboard"
            >
              WeatherDash
            </a>
              
            <a
              href="https://github.com/MBrassey/OpenWeather-Dashboard"
            >
              <i
                class="fab fa-github"
              />
            </a>
          </h3>
          <p>
            Weather dashboard concept using OpenWeather API, ipinfo.io API & jQuery/JS/HTML/CSS
          </p>
        </div>
      </div>
      <div
        class="project"
      >
        <img
          alt="ThoughtAPI"
          class="project-bg"
          src="ThoughtAPI.png"
        />
        <div
          class="project-text"
        >
          <h3>
            <a
              href="https://thoughtstreamapi-mbrassey.herokuapp.com/api/users"
            >
              ThoughtAPI
            </a>
              
            <a
              href="https://github.com/MBrassey/ThoughtStreamAPI"
            >
              <i
                class="fab fa-github"
              />
            </a>
          </h3>
          <p>
            API for a social network web application where users can share their thoughts, react to friends’ thoughts, and create a friend list
          </p>
        </div>
      </div>
      <div
        class="project"
      >
        <img
          alt="TheDevBlog"
          class="project-bg"
          src="TheDevBlog.png"
        />
        <div
          class="project-text"
        >
          <h3>
            <a
              href="https://thedevblog-mbrassey.herokuapp.com"
            >
              TheDevBlog
            </a>
              
            <a
              href="https://github.com/MBrassey/TheDevBlog"
            >
              <i
                class="fab fa-github"
              />
            </a>
          </h3>
          <p>
            Full stack CMS type blog application utilizing ORM with Sequelize/MySQL and adhering to the Model-View-Controller paradigm
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
